learn/cookbooks/indexing_best_practices.mdx

```diff
@@ -0,0 +1,34 @@
+---
+title: Indexing best practices — Meilisearch documentation
+description: All the tips to speed up your documents indexing process.
+---
+
+# Indexing best practices
+
+In this guide, we will review the best practices to help you index your data efficiently and speed up the indexing process.
+
+## Define searchable attributes
+
+To speed up indexing, review your list of [searchable attributes](/learn/configuration/displayed_searchable_attributes#searchable-fields) and ensure it includes only the fields you want to be checked for query word matches. This improves both relevance and search speed by removing irrelevant data from your database. It will also keep your disk usage to the necessary minimum.
+
+### Review filterable and sortable attributes
+
+Some document fields are necessary for [filtering](/learn/fine_tuning_results/filtering) and [sorting](/learn/fine_tuning_results/sorting) results, but they dot not need to be _searchable_. Generally, **numeric and boolean fields** fall into this category. Make sure to review your list of searchable attributes and remove any fields that are only used for filtering or sorting.
+
+## Set ranking rules before indexing
+
+Whenever you update your [ranking rules](/learn/core_concepts/relevancy), Meilisearch will trigger a reindexing of all your documents. This can be a time-consuming process, especially if you have a large dataset. For this reason, it is better to define your ranking rules before indexing your data.
+
+## Optimize document size
+
+Smaller documents are processed faster—so make sure to trim down any unncessary data from your documents. When a document field is missing from the list of [searchable](/reference/api/settings#searchable-attributes), [filterable](/reference/api/settings#filterable-attributes), [sortable](/reference/api/settings#sortable-attributes), or [displayed](/reference/api/settings#displayed-attributes) attributes, it might be best to remove it from the document. To go further, consider compressing your data using methods such as `br`, `deflate`, or `gzip` supported by Meilisearch (see the [supported encoding formats](/reference/api/overview#content-encoding).)
```

Comment:
```suggestion
Smaller documents are processed faster, so make sure to trim down any unncessary data from your documents. When a document field is missing from the list of [searchable](/reference/api/settings#searchable-attributes), [filterable](/reference/api/settings#filterable-attributes), [sortable](/reference/api/settings#sortable-attributes), or [displayed](/reference/api/settings#displayed-attributes) attributes, it might be best to remove it from the document. To go further, consider compressing your data using methods such as `br`, `deflate`, or `gzip`. Consult the [supported encoding formats reference](/reference/api/overview#content-encoding).
```